src/engine/sim_engine/pool.rs

```diff
@@ -420,27 +463,33 @@ impl Pool for SimPool {
             }
         };
 
-        if encryption_info.clevis_info().is_none() {
-            return Err(StratisError::Msg(
-                "This device is not bound to Clevis; refusing to remove the only \
-                unlocking method"
-                    .to_string(),
-            ));
-        }
+        let t = token_slot
+            .ok_or_else(|| StratisError::Msg("Token slot required for V2 pools".to_string()))?;
 
-        Ok(if encryption_info.key_description().is_some() {
-            self.clear_key_desc();
-            DeleteAction::Deleted(Key)
+        if let Some(UnlockMechanism::KeyDesc(_)) = encryption_info.get_info(t) {
+            return Err(StratisError::Msg(format!(
+                "Cannot unbind token slot {t} from Clevis; bound to keyring",
+            )));
+        };
+
+        let slot = encryption_info.remove(t);
+        Ok(if slot {
+            DeleteAction::Deleted(Clevis)
         } else {
             DeleteAction::Identity
         })
     }
 
     fn rebind_keyring(
         &mut self,
+        token_slot: Option<u32>,
         new_key_desc: &KeyDescription,
     ) -> StratisResult<RenameAction<Key>> {
-        let encryption_info = match pool_enc_to_enc!(self.encryption_info()) {
+        let token_slot = token_slot.ok_or_else(|| {
```

Comment:
I believe that you can leave this check out and the sim engine will align better w/ what the strat engine does. As far as I can tell, the strat engine, if given no token slot, will grab the key description associated with the top token slot and do an idempotency check against that. This seems consistent w/ our other decisions.